### PR TITLE
[DOCS] Reformats cluster node stats API

### PR DESCRIPTION
**Pull request by szabosteve**
Relates to elastic/docs#937 and https://github.com/elastic/elasticsearch/issues/45197.

This PR updates the cluster node stats API to align with the new [API reference template](https://github.com/elastic/docs/blob/master/shared/api-ref-ex.asciidoc).

Resources:
* [Cluster node stats API spec](https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/nodes.stats.json)
* [Cluster node stats API doc](https://www.elastic.co/guide/en/elasticsearch/reference/7.3/cluster-nodes-stats.html)

### CODE REVIEW

File: docs/reference/cluster/nodes-stats.asciidoc [hunk 1]
```diff
@@ -1,353 +1,442 @@
...
+
+`GET /_nodes/stats/{metric}/{index_metric}` +
+
+`GET /_nodes/{node_id}/stats/{metric}/{index_metric}`
+
+
+[[cluster-nodes-stats-api-desc]]
+==== {api-description-title}
+
+The cluster nodes stats API allows to retrieve one or more (or all) of the
```
Comment (jrodewig) [inline docs/reference/cluster/nodes-stats.asciidoc]:
```suggestion
You can use the cluster nodes stats API to retrieve statistics for nodes in a cluster.

```

File: docs/reference/cluster/nodes-stats.asciidoc [hunk 2]
```diff
@@ -1,353 +1,442 @@
...
+`GET /_nodes/stats/{metric}/{index_metric}` +
+
+`GET /_nodes/{node_id}/stats/{metric}/{index_metric}`
+
+
+[[cluster-nodes-stats-api-desc]]
+==== {api-description-title}
+
+The cluster nodes stats API allows to retrieve one or more (or all) of the 
+cluster nodes statistics.
```
Comment (jrodewig) [inline docs/reference/cluster/nodes-stats.asciidoc]:
```suggestion

```

File: docs/reference/cluster/nodes-stats.asciidoc [hunk 3]
```diff
@@ -1,353 +1,442 @@
...
+  `thread_pool`::
+      Statistics about each thread pool, including current size, queue and 
+      rejected tasks.
+
+  `transport`::
+      Transport statistics about sent and received bytes in cluster 
+      communication.
+--
+
+`index_metric`::
```
Comment (jrodewig) [inline docs/reference/cluster/nodes-stats.asciidoc]:
```suggestion
`{index_metric}`::
```

**Comment by szabosteve** [comment]
Backported to 7.x and 7.3.

**Pull request merged**